juju/juju
# Rework model get, set, and unset per new spec

Update tests accordingly.

Refs: juju-model-config-collapse

QA:
1. All unit tests pass
2. Live tested:

1. juju bootstrap reed-aws-west-1 awstest
2. juju add-model east us-east-1
3. juju model-config ftp-proxy=bar no-proxy=bar image-stream=videodrome  # set multiple verify the keys are set 
4. juju model-config ftp-proxy # check one  it matches above 
5. juju model-config --reset ftp-proxy no-proxy image-stream # reset multiple
6. juju model-config ftp-proxy=bar # set one and check it matches
7. juju model-config --reset ftp-proxy # reset one
8. juju model-config # makes sure we're back to original
9. juju kill-controller reed-aws-west-1 -y 
10. 

(Review request: http://reviews.vapour.ws/r/5589/)
